packages/widget-v2/src/hooks/useFetchBalances.ts

```diff
@@ -1,20 +1,22 @@
 import { sourceAssetAtom } from "@/state/swapPage";
 import { useAtomValue, useSetAtom } from "jotai";
 import { useGetAccount } from "./useGetAccount";
-import { skipBalancesRequestAtom } from "@/state/balances";
+import { skipAllBalancesRequestAtom, skipBalancesAtom } from "@/state/balances";
 import { useEffect } from "react";
 import { skipAssetsAtom } from "@/state/skipClient";
 
-export const useSourceAccount = () => {
+export const useFetchBalances = () => {
   const sourceAsset = useAtomValue(sourceAssetAtom);
   const getAccount = useGetAccount();
   const { data: assets } = useAtomValue(skipAssetsAtom);
-  const setSkipBalancesRequest = useSetAtom(skipBalancesRequestAtom);
+  const setSkipAllBalancesRequest = useSetAtom(skipAllBalancesRequestAtom);
+  const skipBalances = useAtomValue(skipBalancesAtom);
 
   useEffect(() => {
     const allBalancesRequest = assets?.reduce((acc, asset) => {
       const address = getAccount(asset.chainID)?.address;
       if (address) {
+        if (asset.denom === sourceAsset?.denom) return acc;
```

Comment:
we're omitting the sourceAsset denom, since we're assuming it's being fetched by `useFetchSourceBalance`, honestly it's kinda unnecessary to do this it only prevents one extra denom balance from being fetched again